stable/kubernetes-dashboard/templates/secret.yaml

```diff
@@ -9,3 +9,26 @@ metadata:
   name: {{ template "kubernetes-dashboard.fullname" . }}
   namespace: {{ .Release.Namespace }}
 type: Opaque
+
+---
+
+{{- if and .Values.ingress.tls }}
+{{- if not .Values.ingress.tls.secretName }}
+{{- $ca := genCA "kubernetes-dashboard-ca" 365 }}
+{{- $cert := genSignedCert .Values.ingress.tls.hosts.name nil (list .Values.ingress.tls.hosts.name) 365 $ca }}                                                               
```

Comment:
I've always throught this function was clever, but haven't ever understood the complete ramifications of it. As its generating it on each call, on a helm upgrade will it change? If its self signed and changes, the user will be prompted on every upgrade?

Reply:
I've been using cert-manager for a lot of things recently. It too has support for self signed certs, but also self signed ca's and legit cert integration like with letsencrypt. I haven't yet had a chance to add support for it, but was planning on looking at it eventually.

It would be good to consider the differences and make sure either this proposed api could be switched out for cert-manager, or at least the api would allow them to peacefully co-exist. Having letsencrypt integration would be very valuable to the chart I think.

Reply:
> I've always throught this function was clever, but haven't ever understood the complete ramifications of it. As its generating it on each call, on a helm upgrade will it change? If its self signed and changes, the user will be prompted on every upgrade?

I've not thought about that. I guess you are right.
I'll check it out later.

Reply:
I noticed today that there is a 0.7.0 release of cert-manager now.

Previously you needed to manually upload a tls secret to establish a ca that you could then use to setup the certs automatically.

https://github.com/jetstack/cert-manager/blob/master/docs/getting-started/webhook.rst#how-it-works implies its possible to easily do fully automatically now.